variables.tf

```diff
@@ -32,88 +44,105 @@ variable "allowed_ssh_ranges" {
+    service_prefixes = List of additional prefixes to the dns_name to create
+  }
+  EOF
   type = object({
-    enabled          = bool
-    name             = string
-    dns_name         = string
-    service_prefixes = set(string)
+    enabled          = optional(bool, false)
+    name             = optional(string, "")
+    dns_name         = optional(string, "")
+    service_prefixes = optional(set(string), [])
+
   })
-  default = {
-    enabled          = false
-    name             = ""
-    dns_name         = ""
-    service_prefixes = []
-  }
-  description = "Cloud DNS zone"
+  default = {}
 }
 
-variable "enable_network_policy" {
-  description = "Enable network policy switch"
-  type        = bool
-  default     = true
-}
+variable "kms" {
+  description = <<EOF
+  kms = {
+    database_encryption_key_name = Use an existing KMS key for the Application-layer Secrets Encryption settings. (Optional)
+  }
+  EOF
 
-variable "enable_vertical_pod_autoscaling" {
-  type        = bool
-  default     = true
-  description = "Enable GKE vertical scaling"
-}
+  type = object({
+    database_encryption_key_name = optional(string, null)
+  })
 
-variable "gke_release_channel" {
-  type        = string
-  default     = "STABLE"
-  description = "GKE K8s release channel for master"
+  default = {}
 }
 
-variable "location" {
-  type        = string
-  default     = "us-west1-b"
-  description = "The location (region or zone) of the cluster. A zone creates a single master. Specifying a region creates replicated masters accross all zones"
-}
+variable "gke" {
+  description = <<EOF
+  gke = {
+    k8s_version = Cluster k8s version
+    release_channel = GKE release channel
+    public_access = {
+      enabled = Enable API public endpoint
+      cidrs = List of CIDR ranges permitted for accessing the public endpoint
+    }
+    control_plane_ports =  Firewall ports to open from the master, e.g., webhooks
+    advanced_datapath = Enable the ADVANCED_DATAPATH provider
+    network_policies = Enable network policy switch. Cannot be enabled when enable_advanced_datapath is true
+    vertical_pod_autoscaling = Enable GKE vertical scaling
+    kubeconfig = {
+      path = Specify where the cluster kubeconfig file should be generated.
+    }
+  }
+  EOF
 
-variable "master_firewall_ports" {
-  type        = list(string)
-  default     = []
-  description = "Firewall ports to open from the master, e.g., webhooks"
-}
+  type = object({
+    k8s_version     = optional(string, ""),
+    release_channel = optional(string, "RAPID"),
+    public_access = optional(object({
+      enabled = optional(bool, false),
+      cidrs   = optional(list(string), [])
+    }), {}),
+    control_plane_ports      = optional(list(string), [])
+    advanced_datapath        = optional(bool, true),
+    network_policies         = optional(bool, false),
+    vertical_pod_autoscaling = optional(bool, true),
+    kubeconfig = optional(object({
+      path = optional(string, null)
```

Comment:
what else would you expect here?

Reply:
it matches [EKS(https://github.com/dominodatalab/terraform-aws-eks/blob/main/modules/eks/variables.tf#L127), which has an `extra_args`